syntax = "proto3";

service AppleTree {
    rpc ShakeAppleTree(Time) returns (NumberOfApples) {}
}


message Time {
    // time in minutes
    int32 time_min = 1;
}

message NumberOfApples {
    int32 num_apples = 1;
}

enum IntensityLevel {
    STRONG = 0;
    MEDIUM = 1;
    GENTLE = 2;
}

message Intensity {
    IntensityLevel intensity_level = 1;
}